otter/test/integration/test_lib.py

```diff
@@ -0,0 +1,137 @@
+"""
+Tests for the lib functions for convergence black-box testing.
+"""
+
+from twisted.internet.defer import Deferred
+from twisted.trial.unittest import SynchronousTestCase
+
+from otter.test.integration import lib
+
+
+class IdentityV2Tests(SynchronousTestCase):
+    """Tests for the :class:`IdentityV2` class."""
+
+    def test_missing_params(self):
+        """Missing parameters to the IdentityV2 constructor should raise
+        a ValueError exception.
+        """
+        def x(u, p, e):
+            self.assertRaises(ValueError, lib.IdentityV2, None, u, p, e)
+        x("", "pw", "ep")
+        x("un", "", "ep")
+        x("un", "pw", "")
+        x(None, "pw", "ep")
+        x("un", None, "ep")
+        x("un", "pw", None)
+
+    def test_pool_kwarg(self):
+        """The pool keyword argument should be passed through to the
+        authenticate_user Otter function.
+        """
+        rcs = lib.TestResources()
+
+        class Stub(object):
+            def __init__(self):
+                self.pool = None
+
+            def authenticate_user(self, endpt, user, passwd, pool=None):
+                self.pool = pool
+                return Deferred().addCallback(lambda: {})
+
+        stub = Stub()
+        lib.IdentityV2(
+            auth=stub, username="username",
+            password="password", endpoint="endpoint"
+        ).authenticate_user(rcs)
```

Comment:
Non-blocking:  Possibly authenticate user doesn't need to be called here, since this is just testing the constructor?  Unless the intent is to test that `authenticate_user` does not override the pool, in which case `authenticate_user` never actually finishes because that `Deferred` is never callbacked.

Also handy:  Unless you want to specifically have the Deferred have no result yet (e.g. you are making an `assertNoResult` somewhere), using `defer.succeed({})` might be easier because you don't have to manually callback the Deferred.  It's shorthand for `d = Deferred(); d.callback({})`  :)